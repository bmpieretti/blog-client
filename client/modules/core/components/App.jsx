import React, { Component } from 'react';

export default class App extends Component {
// static propTypes = {}
// static defaultProps = {};

  render() {
    return (
        <h1>Hello World</h1>
    );
  }
}
