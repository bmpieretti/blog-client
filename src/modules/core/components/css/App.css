.helix-plus-client {
  height: 100%;
}