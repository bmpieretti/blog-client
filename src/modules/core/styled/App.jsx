import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const Main = styled.main`
  height: 100%;
`;
