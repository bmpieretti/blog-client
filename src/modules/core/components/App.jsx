import React from 'react';
import ErrorBoundary from './ErrorBoundary';
import { Main } from '../styled/App';

const App = () => (
  <ErrorBoundary>
    <Main />
    <div> Hello World! </div>
  </ErrorBoundary>
);

export default App;
