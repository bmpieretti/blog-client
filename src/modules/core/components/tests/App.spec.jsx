import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from '../App';

describe('App Component: ', () => {
  test('should render', () => {
    const component = shallow(<App />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  test('should render main and styles', () => {
    const mainElement = shallow(<App />).dive().dive();
    expect(shallowToJson(mainElement)).toMatchSnapshot();
  });
});
