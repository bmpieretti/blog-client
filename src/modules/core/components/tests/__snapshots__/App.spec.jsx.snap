// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component:  should render 1`] = `
<ErrorBoundary>
  <App__Main />
</ErrorBoundary>
`;

exports[`App Component:  should render main and styles 1`] = `
.c0 {
  height: 100%;
}

<main
  className="c0"
/>
`;
